From 6d3888fc5ab106aa1ed249baee62ae6d5e1712b2 Mon Sep 17 00:00:00 2001
From: Michal Wasko <michal.wasko@intel.com>
Date: Wed, 10 Oct 2018 13:27:41 +0200
Subject: [PATCH 1201/1214] ASoC: Intel: Skylake: Set DUM bit in EM2 register
 on skl resume

Issue fix the IP bug of incorrect position
reporting for capture stream

Change-Id: Ic183a8e74eef7624c35003273f6ec46e6d7c3d9f
Tracked-On: PKT-3663, OAM-83914
Signed-off-by: Michal Wasko <michal.wasko@intel.com>
---
 sound/soc/intel/skylake/skl.c | 33 ++++++++++++++++-----------------
 1 file changed, 16 insertions(+), 17 deletions(-)

diff --git a/sound/soc/intel/skylake/skl.c b/sound/soc/intel/skylake/skl.c
index c16ad3b..1aa314f 100644
--- a/sound/soc/intel/skylake/skl.c
+++ b/sound/soc/intel/skylake/skl.c
@@ -58,21 +58,6 @@ static void skl_update_pci_byte(struct pci_dev *pci, unsigned int reg,
 	pci_write_config_byte(pci, reg, data);
 }
 
-static void skl_init_pci(struct skl *skl)
-{
-	struct hdac_bus *bus = skl_to_bus(skl);
-
-	/*
-	 * Clear bits 0-2 of PCI register TCSEL (at offset 0x44)
-	 * TCSEL == Traffic Class Select Register, which sets PCI express QOS
-	 * Ensuring these bits are 0 clears playback static on some HD Audio
-	 * codecs.
-	 * The PCI register TCSEL is defined in the Intel manuals.
-	 */
-	dev_dbg(bus->dev, "Clearing TCSEL\n");
-	skl_update_pci_byte(skl->pci, AZX_PCIREG_TCSEL, 0x07, 0);
-}
-
 static void update_pci_dword(struct pci_dev *pci,
 			unsigned int reg, u32 mask, u32 val)
 {
@@ -241,6 +226,22 @@ static void skl_dum_set(struct hdac_bus *bus)
 	snd_hdac_chip_writel(bus, VS_EM2, (reg | AZX_EM2_DUM_MASK));
 }
 
+static void skl_init_pci(struct skl *skl)
+{
+	struct hdac_bus *bus = skl_to_bus(skl);
+
+	/*
+	* Clear bits 0-2 of PCI register TCSEL (at offset 0x44)
+	* TCSEL == Traffic Class Select Register, which sets PCI express QOS
+	* Ensuring these bits are 0 clears playback static on some HD Audio
+	* codecs.
+	* The PCI register TCSEL is defined in the Intel manuals.
+	*/
+	dev_dbg(bus->dev, "Clearing TCSEL\n");
+	skl_update_pci_byte(skl->pci, AZX_PCIREG_TCSEL, 0x07, 0);
+	skl_dum_set(bus);
+}
+
 /* called from IRQ */
 static void skl_stream_update(struct hdac_bus *bus, struct hdac_stream *hstr)
 {
@@ -1010,8 +1011,6 @@ static int skl_first_init(struct hdac_bus *bus)
 	/* initialize chip */
 	skl_init_pci(skl);
 
-	skl_dum_set(bus);
-
 	return skl_init_chip(bus, true);
 }
 
-- 
2.7.4

